api/src/test/java/org/pmiops/workbench/ras/RasLinkServiceTest.java

```diff
@@ -0,0 +1,167 @@
+import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest;
+import org.springframework.boot.test.context.TestConfiguration;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Import;
+import org.springframework.context.annotation.Scope;
+import org.springframework.test.context.junit4.SpringRunner;
+
+@RunWith(SpringRunner.class)
+@DataJpaTest
+public class RasLinkServiceTest {
+  private static final Timestamp NOW = Timestamp.from(Instant.now());
+  private static final FakeClock CLOCK = new FakeClock(NOW.toInstant(), ZoneId.systemDefault());
+
+  private static final String AUTH_CODE = "code";
+  private static final String REDIRECT_URL = "url";
+  private static final String ACCESS_TOKEN = "access_token_1";
+  private static final String LOGIN_GOV_USERNAME = "foo@gmail.com";
+  private static final String USER_INFO_JSON_LOGIN_GOV =
+      "{\"preferred_username\":\"user1@Login.Gov.com\", \"email\":\"" + LOGIN_GOV_USERNAME + "\"}";
+  private static final String USER_INFO_JSON_ERA =
+      "{\"preferred_username\":\"user2@eraCommons.com\", \"email\":\"" + LOGIN_GOV_USERNAME + "\"}";
+  private static final String ID_TOKEN_JWT_IAL_1 =
+      JWT.create()
+          .withClaim(
+              ACR_CLAIM,
+              "https://stsstg.nih.gov/assurance/ial/1 https://stsstg.nih.gov/assurance/aal/1")
+          .sign(Algorithm.none());
+  private static final String ID_TOKEN_JWT_IAL_2 =
+      JWT.create()
+          .withClaim(
+              ACR_CLAIM,
+              "https://stsstg.nih.gov/assurance/ial/2 https://stsstg.nih.gov/assurance/aal/1")
+          .sign(Algorithm.none());
+  private static final TokenResponse TOKEN_RESPONSE_IAL1 =
+      new TokenResponse().setAccessToken(ACCESS_TOKEN).set(Id_TOKEN_FIELD_NAME, ID_TOKEN_JWT_IAL_1);
+  private static final TokenResponse TOKEN_RESPONSE_IAL2 =
+      new TokenResponse().setAccessToken(ACCESS_TOKEN).set(Id_TOKEN_FIELD_NAME, ID_TOKEN_JWT_IAL_2);
+
+  private long userId;
+  private static DbUser currentUser;
+
+  private RasLinkService rasLinkService;
+  private ObjectMapper objectMapper = new ObjectMapper();
+
+  @Autowired private UserService userService;
+  @Autowired private UserDao userDao;
+  @Mock private OpenIdConnectClient mockOidcClient;
+
+  @TestConfiguration
+  @Import({
+    UserServiceTestConfiguration.class,
+  })
+  static class Configuration {
+    @Bean
+    Clock clock() {
+      return CLOCK;
+    }
+
+    @Bean
+    Random random() {
+      return new FakeLongRandom(123);
+    }
+
+    @Bean
+    @Scope(ConfigurableBeanFactory.SCOPE_PROTOTYPE)
+    public WorkbenchConfig getWorkbenchConfig() {
+      return WorkbenchConfig.createEmptyConfig();
+    }
+
+    @Bean
+    @Scope("prototype")
+    DbUser user() {
+      return currentUser;
+    }
+  }
+
+  @MockBean UserServiceAuditor mockUserServiceAuditor;
```

Comment:
Why did you need these? If it's for `UserService`, shouldn't those be documented here as required deps under `UserServiceTestConfiguration` - otherwise it's a bit misleading if that wasn't actually exhaustive (I'm not familiar with the class so I don't have the context).

Reply:
I am confused about this as well, and it took me a while to figure out what's going on. Those are all UserService dependencies, but all other tests like:
https://github.com/all-of-us/workbench/blob/master/api/src/test/java/org/pmiops/workbench/api/ConceptSetsControllerTest.java
https://github.com/all-of-us/workbench/blob/master/api/src/test/java/org/pmiops/workbench/api/RuntimeControllerTest.java
https://github.com/all-of-us/workbench/blob/master/api/src/test/java/org/pmiops/workbench/api/DataSetControllerTest.java
(basicly **all** tests that use UserServiceTestConfiguration are the same)
all have this pattern. I guess that is because UserServiceTestConfiguration hold them most common, simple dependencies, while each test will in charge of creating other beans, depends on whether they need mock or not? 
So I just follow other's pattern and follow the common pattern.